pkg/ccl/importccl/exportparquet.go

```diff
@@ -181,57 +210,356 @@ func newParquetColumn(typ *types.T, name string, nullable bool) (parquetColumn,
+			return []byte(d.(*tree.DGeometry).EWKB()), nil
 		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			g, err := geo.ParseGeometryFromEWKBUnsafe(geopb.EWKB(x.([]byte)))
+			if err != nil {
+				return nil, err
+			}
+			return &tree.DGeometry{Geometry: g}, nil
+		}
+	case types.DateFamily:
+		// Even though the parquet vendor supports Dates, we export Dates as strings
+		// because the vendor only supports encoding them as an int32, the Days
+		// since the Unix epoch, which according CRDB's `date.UnixEpochDays( )` (in
+		// pgdate package) is vulnerable to overflow.
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			date := d.(*tree.DDate)
+			ds := roundtripStringer(date)
+			return []byte(ds), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			dStr := string(x.([]byte))
+			d, dependCtx, err := tree.ParseDDate(nil, dStr)
+			if dependCtx {
+				return nil, fmt.Errorf("decoding date %s failed. depends on context", string(x.([]byte)))
+			}
+			return d, err
+		}
+	case types.TimeFamily:
+		schemaEl.Type = parquet.TypePtr(parquet.Type_INT64)
+		schemaEl.LogicalType = parquet.NewLogicalType()
+		schemaEl.LogicalType.TIME = parquet.NewTimeType()
+		t := parquet.NewTimeUnit()
+		t.MICROS = parquet.NewMicroSeconds()
+		schemaEl.LogicalType.TIME.Unit = t
+		schemaEl.LogicalType.TIME.IsAdjustedToUTC = true // per crdb docs
 
-	case types.FloatFamily:
-		if typ.Oid() == oid.T_float8 {
-			col.definition.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
-			col.encodeFn = func(d tree.Datum) (interface{}, error) {
-				return float64(*d.(*tree.DFloat)), nil
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			// Time of day is stored in microseconds since midnight,
+			// which is also how parquet stores time
+			time := d.(*tree.DTime)
+			m := int64(*time)
+			return m, nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.MakeDTime(timeofday.TimeOfDay(x.(int64))), nil
+		}
+	case types.TimeTZFamily:
+		// The parquet vendor does not support an efficient encoding of TimeTZ
+		// (i.e. a datetime field and a timezone field), so we must fall back to
+		// encoding the whole TimeTZ as a string.
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DTimeTZ).TimeTZ.String()), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			d, dependsOnCtx, err := tree.ParseDTimeTZ(nil, string(x.([]byte)), time.Microsecond)
+			if dependsOnCtx {
+				return nil, fmt.Errorf("parsed ime depends on context")
+			}
+			return d, err
+		}
+	case types.IntervalFamily:
+		// The parquet vendor actually supports intervals as a parquet converted type,
+		// but converted types have been deprecated in the Apache Parquet format.
+		// https://github.com/fraugster/parquet-go#supported-converted-types
```

Comment:
Interesting, the spec also has a logical type for this: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#interval but it doesn't seem to be supported by the library.

Reply:
Indeed. So I don't think we can do anything differently.